experimental/AlgebraicStatistics/src/GaussianGraphicalModels.jl

```diff
@@ -0,0 +1,412 @@
+[s[1, 3]   s[2, 3]   s[3, 3]]
+```
+"""
+function covariance_matrix(R::GaussianRing)
+
+    R.covariance_matrix
+end
+
+
+
+###################################################################################
+#
+#       Directed Gaussian Graphical Models
+#
+###################################################################################
+
+
+@doc raw"""
+    graphical_model(G::Graph{Directed}, S::GaussianRing, l_var_name::String="l")
+
+A parametric statistical model associated to a directed acyclic graph.
+It contains a directed acylic graph `G`, a GaussianRing `S` where the vanishing ideal of the model naturally lives, 
+and a parameter ring whose variables `l[i,j]` correspond to the directed edges `i -> j` in `G`. 
+
+## Examples
+
+``` jldoctest directed_ggm
+julia> M = graphical_model(graph_from_edges(Directed, [[1,2], [2,3]]), gaussian_ring(3))
+Gaussian graphical model on a directed graph with edges:
+(1, 2), (2, 3)
+```
+"""
+function graphical_model(G::Graph{Directed}, S::GaussianRing, l_var_name::String="l")::GraphicalModel
+
+    R, l, w = polynomial_ring(QQ, ["l"*string([src(e), dst(e)]) for e in edges(G)], ["w"*string([v]) for v in vertices(G)])
+
+    l = Dict([(Tuple(var_index(x)), x) for x in l])
+
+    GraphicalModel(G, S, R, [l, w])
+end
+
+
+function show(io::IO, M::GraphicalModel{Graph{Directed}, GaussianRing})
+
+    G = M.graph
+    E = [(src(e), dst(e)) for e in edges(G)]
+
+    if length(E) == 0
+        print("Directed Gaussian graphical model on the empty graph")
+    end
+
+    if length(E) > 0
+        print("Gaussian graphical model on a directed graph with edges:", "\n", string(E)[2:end-1])
+    end
+end
+
+
+
+@doc raw"""
+    directed_edges_matrix(M::GraphicalModel{Graph{Directed}, GaussianRing})
+
+Creates the weighted adjacency matrix $\Lambda$ of a directed graph `G` whose entries are the parameter ring of the graphical model `M`.
+
+## Examples
+
+``` jldoctest directed_ggm
+julia> directed_edges_matrix(M)
+[0   l[1, 2]         0]
+[0         0   l[2, 3]]
+[0         0         0]
+```
+"""
+function directed_edges_matrix(M::GraphicalModel{Graph{Directed}, GaussianRing})
+
+    G = graph(M)
+    l = param_gens(M)[1]
+    L = matrix(M.param_ring, [[has_edge(G, i, j) ? l[i,j] : 0 for j in vertices(G)] for i in vertices(G)])
+end
+
+
+@doc raw"""
+    error_covariance_matrix(M::GraphicalModel{Graph{Directed}, GaussianRing})
+
+Creates the covariance matrix $ \Omega $ of the independent error terms in a directed Gaussian graphical model `M`
+
+## Examples
+
+``` jldoctest directed_ggm
+julia> error_covariance_matrix(M)
+[w[1]      0      0]
+[   0   w[2]      0]
+[   0      0   w[3]]
+```
+"""
+function error_covariance_matrix(M::GraphicalModel{Graph{Directed}, GaussianRing})
+
+    G = graph(M)
+    w = param_gens(M)[2]
+    W = matrix(M.param_ring, [[i == j ? w[i] : 0 for j in vertices(G)] for i in vertices(G)])
+end
+
+
+
+@doc raw"""
+    parameterization(M::GraphicalModel{Graph{Directed}, GaussianRing})
```

Comment:
while both "parametrization" and "parameterization" are valid spellings, elsewhere we already use `parametrization` so I would suggest following that (CC @JohnAAbbott and @mjrodgers just in case our native speakers have a different view on this)
```suggestion
    parameterization(M::GraphicalModel{Graph{Directed}, GaussianRing})
```

Reply:
Done.